src/Task/Infection.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace GrumPHP\Task;
+
+use GrumPHP\Runner\TaskResult;
+use GrumPHP\Task\Context\ContextInterface;
+use GrumPHP\Task\Context\GitPreCommitContext;
+use GrumPHP\Task\Context\RunContext;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Infection task
+ */
+class Infection extends AbstractExternalTask
+{
+    /**
+     * @return string
+     */
+    public function getName()
+    {
+        return 'infection';
+    }
+
+    /**
+     * @return OptionsResolver
+     */
+    public function getConfigurableOptions()
+    {
+        $resolver = new OptionsResolver();
+
+        $resolver->setDefaults([
+            'threads' => null,
+            'test_framework' => null,
+            'only_covered' => false,
+            'configuration' => null,
+            'min_msi' => null,
+            'min_covered_msi' => null,
+            'mutators' => [],
+            'triggered_by' => ['php'],
+        ]);
+
+        $resolver->addAllowedTypes('threads', ['null', 'int']);
+        $resolver->addAllowedTypes('test_framework', ['null', 'string']);
+        $resolver->addAllowedTypes('only_covered', ['bool']);
+        $resolver->addAllowedTypes('configuration', ['null', 'string']);
+        $resolver->addAllowedTypes('min_msi', ['null', 'integer']);
+        $resolver->addAllowedTypes('min_covered_msi', ['null', 'integer']);
+        $resolver->addAllowedTypes('mutators', ['array']);
+        $resolver->addAllowedTypes('triggered_by', ['array']);
+
+        return $resolver;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function canRunInContext(ContextInterface $context)
+    {
+        return ($context instanceof GitPreCommitContext || $context instanceof RunContext);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function run(ContextInterface $context)
+    {
+        $config = $this->getConfiguration();
+        $files = $context->getFiles()->extensions($config['triggered_by']);
+
+        if (0 === count($files)) {
+            return TaskResult::createSkipped($this, $context);
+        }
+
+        $arguments = $this->processBuilder->createArgumentsForCommand('infection');
```

Comment:
Infection requires xdebug or phpdbg to be installed, so I think this task should be skipped if at least one of those extensions is not installed

Reply:
I don't think grumphp is responsible to check that. If you don't have xdebug or phpdbg enabled, your tests will not run anyway.